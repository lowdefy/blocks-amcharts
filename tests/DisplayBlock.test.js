/*
  Copyright 2020 Lowdefy, Inc

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
*/

import React from 'react';
import { mockBlock, runBlockSchemaTests, runRenderTests } from '@lowdefy/block-tools';
import { configure, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
configure({ adapter: new Adapter() });

import { DisplayBlock } from '../src';
import examples from '../demo/examples/DisplayBlock.yaml';
import meta from '../src/blocks/DisplayBlock/DisplayBlock.json';

runRenderTests({ examples, Block: DisplayBlock, meta });
runBlockSchemaTests({ examples, meta });

const { before, methods, getProps } = mockBlock({ meta });
beforeEach(before);

test('callAction onClick', () => {
  const block = {
    id: 'one',
    type: 'DisplayBlock',
  };

  const Shell = () => <DisplayBlock {...getProps(block)} methods={methods} />;
  const wrapper = mount(<Shell />);
  wrapper.find('[data-testid="one"]').simulate('click');
  expect(methods.callAction).toHaveBeenCalledWith({ action: 'onClick' });
});
