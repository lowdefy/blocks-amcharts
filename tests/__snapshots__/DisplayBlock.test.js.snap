// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render actions.onClick - value[0] 1`] = `
<div
  className="{\\"style\\":[{\\"outline\\":\\"none\\",\\"cursor\\":\\"pointer\\"},null]}"
  data-testid="actions.onClick"
  id="actions.onClick"
  onClick={[Function]}
>
  New Lowdefy display block 
</div>
`;

exports[`Render default - value[0] 1`] = `
<div
  className="{\\"style\\":[{\\"outline\\":\\"none\\"},null]}"
  data-testid="default"
  id="default"
  onClick={[Function]}
>
  New Lowdefy display block 
</div>
`;

exports[`Render property.style - value[0] 1`] = `
<div
  className="{\\"style\\":[{\\"outline\\":\\"none\\"},{\\"color\\":\\"red\\"}]}"
  data-testid="property.style"
  id="property.style"
  onClick={[Function]}
>
  New Lowdefy display block 
</div>
`;

exports[`Render property.title - value[0] 1`] = `
<div
  className="{\\"style\\":[{\\"outline\\":\\"none\\"},null]}"
  data-testid="property.title"
  id="property.title"
  onClick={[Function]}
>
  New Lowdefy display block 
  with title: Foo
</div>
`;

exports[`Test Schema actions.onClick 1`] = `true`;

exports[`Test Schema actions.onClick 2`] = `null`;

exports[`Test Schema default 1`] = `true`;

exports[`Test Schema default 2`] = `null`;

exports[`Test Schema property.style 1`] = `true`;

exports[`Test Schema property.style 2`] = `null`;

exports[`Test Schema property.title 1`] = `true`;

exports[`Test Schema property.title 2`] = `null`;
